pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -799,6 +799,12 @@ func getURIRank(match *istio.HTTPMatchRequest) int {
 	if match.Uri == nil {
 		return -1
 	}
+
+	// catch all route should be the lowest priority
+	if isCatchAll(match) {
```

Comment:
special handler for catch all rout
cc @howardjohn

Reply:
This seems odd to.

1. This function is giving the rank of the URI match. Why is a URI match looking at method, headers, query param.
2. Why does sortHTTPRoutes not already account for 'catch all' implicitly? It should make less exact matches lowest priority. Why is that not already happening?

Reply:
> This function is giving the rank of the URI match. Why is a URI match looking at method, headers, query param.

most likely share same logic as `func IsCatchAllRoute(r *route.Route) bool {`

> Why does sortHTTPRoutes not already account for 'catch all' implicitly? It should make less exact matches lowest priority. Why is that not already happening?

The match order from comment is `Exact -> Prefix -> Regex`.
if we didn't handle `catch all`, any `Prefix` or `Regex` match will be ignored when istio start to translator to xds after sorting.